networkx/algorithms/min_edge_cover.py

```diff
@@ -0,0 +1,74 @@
+"""
+An edge cover of a graph is a set of edges such that every vertex of
+the graph is incident to at least one edge of the set.
+A minimum edge cover is an edge covering of smallest possible size.
+
+https://en.wikipedia.org/wiki/Edge_cover
+"""
+#    Copyright (C) 2015 by
+#    Nishant Nikhil <nishantiam@gmail.com>
+#    All rights reserved.
+#    BSD license.
+
+import networkx as nx
+from networkx.algorithms import bipartite
+
+
+def min_edge_cover(G):
+    """
+    A smallest edge cover can be found in polynomial time by finding
+    a maximum matching and extending it greedily so that all vertices
+    are covered.
+
+    Parameters
+    ----------
+    G : NetworkX graph
+        Undirected graph
+        Unweighted graph
+
+    Returns
+    -------
+    min_cover : dictionary
+        The matching is returned as a dictionary, min_weighted_edge_cover,
+        such that min_weighted_edge_cover[v] has a list of nodes so that
+        the edge joining them occurs in minimum edge cover.
+        If no edge cover exists, return an empty dictionary.
+
+    Notes
+    -----
+    Worst Case runtime of maximum_matching for bipartite graph is
+    O(number_of_nodes ** 2.5)
+    Worst Case runtime of maximum_matching for other graphs is
+    O(number_of_nodes ** 3)
+    And the greedy extension works in O(number_of_nodes)
+    So worst case run time of the function is
+    O(number_of_nodes ** 2.5) or O(number_of_nodes ** 3) accordingly.
+    """
+    maximum_matching = {}
+    is_cover_exist = True
+    for node in G.nodes():
+        if not G[node]:
+            is_cover_exist = False
+            break
+    if not is_cover_exist:
+        # min_cover does not exist
+        # so return an empty dictionary
+        return maximum_matching
+    if bipartite.is_bipartite(G):
+        # for better run time
+        maximum_matching = bipartite.maximum_matching(G)
+    else:
+        maximum_matching = nx.max_weight_matching(G, maxcardinality=True)
+    min_cover = {}
+    # min_weighted_edge_cover is superset of maximum_matching
+    # but unlike maximum_matching, here a key can have multiple values
+    for key in maximum_matching.keys():
```

Comment:
It would be simpler to return a set of edges (or an iterator over a set of edges). Then show in the `Examples` section of the documentation how to compute the edges incident to a particular vertex.